pkg/v3/coordinator/log_event_coordinator.go

```diff
@@ -46,15 +46,38 @@ func NewReportCoordinator(logs EventProvider, conf config.OffchainConfig, logger
 	}
 }
 
+func (rc *reportCoordinator) isLogEventUpkeep(upkeep ocr2keepers.ReportedUpkeep) bool {
+	// Checking if Extension is a map
+	extension, ok := upkeep.Trigger.Extension.(map[string]interface{})
+	if !ok {
+		return false
+	}
+
+	// Checking if "txHash" exists and is a string
+	_, ok = extension["txHash"].(string)
+	if !ok {
+		return false
+	}
+
+	// Return true if all checks pass
+	return true
+}
+
 func (rc *reportCoordinator) Accept(upkeep ocr2keepers.ReportedUpkeep) {
+	isLogEvent := rc.isLogEventUpkeep(upkeep)
+	if !isLogEvent {
+		rc.logger.Printf("Upkeep is not log event based, skipping ID: %s", upkeep.ID)
+		return
+	}
```

Comment:
One note on coding style that isn't super important:

I would rewrite this as the following just to avoid introducing another variable into scope and it reads cleaner.
```
if !rc.isLogEventUpkeep(upkeep) {
    // ...
}
```